AlphaWallet/AppCoordinator.swift

```diff
@@ -481,8 +504,76 @@ extension AppCoordinator: AssetDefinitionStoreDelegate {
     }
 }
 
-extension AppCoordinator: UrlSchemeCoordinatorDelegate {
-    func resolve(for coordinator: UrlSchemeCoordinator) -> UrlSchemeResolver? {
+extension AppCoordinator: UniversalLinkCoordinatorDelegate {
+
+    private var hasImportMagicTokenCoordinator: ImportMagicTokenCoordinator? {
+        return coordinators.compactMap { $0 as? ImportMagicTokenCoordinator }.first
+    }
+
+    func handle(url: MagicLinkUrl, for resolver: UrlSchemeResolver) {
+        switch url {
+        case .eip681(let url):
+            let eip681UrlResolver = Eip681UrlResolver(tokensDatastores: resolver.tokensStorages, assetDefinitionStore: assetDefinitionStore)
+            guard let promise = eip681UrlResolver.resolve(url: url) else { return }
+            //NOTE: could be greate if we use loader here, but we can't as promise could be not resolved at all
```

Comment:
Can you elaborate?

Reply:
when selection `.eip681` url it takes some time, to resolve token info like, name, symbol, decimals and so on.  So in this case use need to wait for it. Could be better to show some loader  to display processing